### DISCUSSION TITLE
TypeScript type issues after upgrading to unified v11

### DISCUSSION THREAD

**Discussion by rfgamaral** [remarkjs/remark#1214] [Q&A] [answered]
Hi there,

I'm trying to upgrade unified to v11 (along with a bunch of related dependencies) in [Typist](https://github.com/Doist/typist) ([relevant PR](https://github.com/Doist/typist/pull/395)), but I'm having a few issues with TypeScript types. You can see everything that went wrong in [this GitHub action run](https://github.com/Doist/typist/actions/runs/5881606259/job/15950465922?pr=395), but I'll give a few examples below.

## Custom plugin issue

I have no need for all that the `remark-gfm` plugin provides, I only need strikethrough and autolink literals, and for that I've created 2 custom plugins. They're both very similar, so taking one as an example ([source](https://github.com/Doist/typist/blob/main/src/serializers/html/plugins/remark-strikethrough.ts)):

```ts
import {
    gfmStrikethroughFromMarkdown,
    gfmStrikethroughToMarkdown,
} from 'mdast-util-gfm-strikethrough'
import { gfmStrikethrough } from 'micromark-extension-gfm-strikethrough'

import type { Options } from 'micromark-extension-gfm-strikethrough'
import type { Processor } from 'unified'

/**
 * A remark plugin to add support for the strikethrough extension from the GitHub Flavored Markdown
 * (GFM) specification.
 *
 * This is an standalone plugin which makes use of both the `mdast-util-gfm-strikethrough` and
 * `micromark-extension-gfm-strikethrough` packages, and the implementation is inspired by the
 * third-party `remark-gfm` plugin.
 *
 * The reason why we don't use `remark-gfm` directly is because we don't want to support all other
 * GFM features (footnotes, tables, tagfilter, and tasklists).
 *
 * @param options Configuration options for the plugin.
 */
function remarkStrikethrough(this: Processor, options: Options = {}) {
    const data = this.data()

    function add(field: string, value: unknown) {
        const list = (data[field] ? data[field] : (data[field] = [])) as unknown[]

        list.push(value)
    }

    add('micromarkExtensions', gfmStrikethrough(options))
    add('fromMarkdownExtensions', gfmStrikethroughFromMarkdown())
    add('toMarkdownExtensions', gfmStrikethroughToMarkdown())
}

export { remarkStrikethrough }
```

Here's the issue I'm having:

![image](https://github.com/remarkjs/remark/assets/96476/184a23de-4cf6-438a-aa7b-5c43e63e1f89)

## Third-party plugins issue

I'm initializing `unified` like this:

```ts
const unifiedProcessor = unified().use(remarkParse)
```

And as you can see, I'm using the `remark-parse` plugin, here's the error I'm getting:

![image](https://github.com/remarkjs/remark/assets/96476/17b4aaef-8139-4668-b5b4-70b61c6908fe)

There's a bunch of similar errors for other third-party plugins. Actually, every single third-party plugin used in [this file](https://github.com/Doist/typist/blob/main/src/serializers/html/html.ts) is showing the same error.

In this case, I'm not doing anything fancy and/or out of the ordinary, I'm simply using the plugin, but it seems incompatible with unified v11.

---

Please advise on how to proceed for both these issues.

Much appreciated.

**Comment by wooorm** [comment]
Hey: downgrade. All projects will be updated. It’s 700+ packages, which takes time. Or `ts-expect-error`. It’s only types

**Reply by rfgamaral** [reply]
Thank you for clarifying, we'll wait!

**Comment by ChristianMurphy** [comment, marked as answer]
Welcome @rfgamaral!
Thanks for your excitement and interest trying out the latest version.
Echoing @wooorm, `unified` 11 support is making it's way through all the projects, and has not yet made it to the `remark` projects, it will in the coming weeks.
For now, hold the `unified` version to 10, and keep an eye out for new major versions of `remark` plugins which will support `unified` 11 and include other helpful fixes and features.

**Reply by rfgamaral** [reply]
Thank you for clarifying, we'll wait!
